base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp

```diff
@@ -75,6 +76,49 @@ std::string HumanFriendlyStateName(cvd::InstanceState state) {
   return name;
 }
 
+// Adds more information to the json object returned by cvd_internal_status,
+// including some that cvd_internal_status normally returns but doesn't when the
+// instance is not running.
+void EnrichInstanceJson(const selector::LocalInstanceGroup& group,
+                        const cvd::Instance& instance,
+                        Json::Value& instance_json) {
+  CHECK(instance_json.isMember("instance_name") &&
+        instance_json["instance_name"].isString())
+      << "JSON returned from status fetcher has unexpected format";
+  instance_json["instance_name"] = instance.name();
+  instance_json["status"] = HumanFriendlyStateName(instance.state());
+  instance_json["adb_port"] = instance.adb_port();
+  instance_json["assembly_dir"] = group.AssemblyDir();
+  instance_json["instance_dir"] = group.InstanceDir(instance);
+  instance_json["instance_name"] = instance.name();
+  instance_json["webrtc_device_id"] = instance.webrtc_device_id();
```

Comment:
I don't like "Enrich" that much either, any suggestion is welcome.

This code intentionally overrides fields from `cvd_internal_status` for two reasons:
1. `cvd` knows more about the host than the cuttlefish devices does. For example, `cvd` knows where the host orchestrator is, if that ever changes cvd can easily change at the same time because they are released together, from the same source; the AOSP code can't react to those changes easily.
2. This JSON is `cvd`'s output, it's better to have `cvd` be aware of and enforce its own interface instead of relying on `cvd_internal_status` from who-knows-what-branch. My intention is evolve this to only rely on `cvd_internal_status`'s exit code for the instance and other information that can change while it's running (displays), but for everything that stays constant after boot (ADB, webrtc_id, etc) rely on the instance database, but it'll require more work to get there.

Reply:
`OverrideInstanceJson`?